cardano-cli/test/cardano-cli-golden/Test/Golden/CreateTestnetData.hs

```diff
@@ -134,3 +137,40 @@ hprop_golden_create_testnet_data_transient_stake_delegators =
     -- We just test that the command doesn't crash when we execute a different path.
     -- For the golden part of this test, we are anyway covered by 'hprop_golden_create_testnet_data'
     -- that generates strictly more stuff.
+
+-- Execute this test with:
+-- @cabal test cardano-cli-golden --test-options '-p "/golden create testnet data deleg non deleg/"'@
+hprop_golden_create_testnet_data_deleg_non_deleg :: Property
+hprop_golden_create_testnet_data_deleg_non_deleg =
+  propertyOnce $ moduleWorkspace "tmp" $ \tempDir -> do
+
+    let outputDir = tempDir </> "out"
+        totalSupply :: Int = 2000000000000 -- 2*10^12
+        delegatedSupply :: Int = 500000000000 -- 5*10^11, i.e. totalSupply / 4
+
+    void $ execCardanoCLI
+      [ "conway", "genesis", "create-testnet-data"
+      , "--utxo-keys", "1"
+      , "--total-supply", show totalSupply
+      , "--delegated-supply", show delegatedSupply
+      , "--out-dir", outputDir]
+
+    bs <- liftIO $ LBS.readFile $ outputDir </> "genesis.json"
+    genesis :: ShelleyGenesis StandardCrypto <- Aeson.throwDecode bs
```

Comment:
You can use here https://hackage.haskell.org/package/hedgehog-extras-0.6.1.0/docs/Hedgehog-Extras-Test-File.html#v:readJsonFileOk

Reply:
Done and learnt about `readJsonFileOk` :+1: